services/server/server.go

```diff
@@ -360,6 +360,7 @@ func LoadPlugins(ctx context.Context, config *srvconfig.Config) ([]*plugin.Regis
 
```

Comment:
While `SharedNamespace()` handles invalid combinations implicitly (e.g. the method returns "" when shared is true), I'd like to reject invalid combinations here explicitly to reduce the chance of misconfiguration.